### ISSUE TITLE
wrong cursor position when virtual text positioned at end of line

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#10786]
**Steps to reproduce**

Run this shell command:

    vim -Nu NONE -S <(tee <<'EOF'
        vim9script
        prop_type_add('some_type_name', {
            highlight: 'Question',
        })
        prop_add(1, col('$'), {
            type: 'some_type_name',
            text: ' some virtual text'
        })
    EOF
    )

The cursor is drawn after the virtual text.
Press `i` to enter insert mode, and insert the characters `abc`:  the cursor is still drawn after the virtual text.

**Expected behavior**

The cursor is initially drawn at the start of the line.
When we insert characters, the cursor follows them.  It's drawn between the real text, and the virtual one (not after both).

**Version of Vim**

9.0 Included patches: 1-71

**Environment**

Operating system: Ubuntu 20.04.4 LTS
Terminal: xterm
Value of $TERM: xterm-256color
Shell: GNU bash, version 5.0.17

**Additional context**

N/A

**Comment by brammool** [comment]
Virtual text at the end of the line isn't implemented yet.  Feel free to have a look into how this should work.

**Comment by bfrg** [comment]
> Virtual text at the end of the line isn't implemented yet.

It's not just at the end of the line where it doesn't work yet.

The following example will create a new buffer with `12345678` in the first line, and add a virtual text in between the 4 and 5.
```vim
vim9script
edit foobar
setline(1, '12345678')
prop_type_add('test', {highlight: 'Special'})
prop_add(1, 5, {
    type: 'test',
    text: ' virtual text '
})
```
Place the cursor at the 4 and then hit `a` and start inserting text. As soon as you press `a`, the cursor jumps to the end of the virtual text (in the above example it jumps to the 5). The first character is inserted after the 4 where it's supposed to be. After that the cursor jumps to the end of the line and further text is inserted there.

**Comment by bfrg** [comment]
@brammool This is what line annotations looks like in qt creator:

https://user-images.githubusercontent.com/6266600/181498152-c27066a1-5949-46f3-8317-184911362fe7.mp4

There are 4 ways to align line annotations in qt creator. See [Specifying Line Annotation Positions](https://doc.qt.io/qtcreator/creator-checking-code-syntax.html#specifying-line-annotation-positions) for all the options.

**Comment by lacygoill** [comment]
Fixed in patch [9.0.0121](https://github.com/vim/vim/releases/tag/v9.0.0121):
```vim
vim9script
prop_type_add('some_type_name', {
    highlight: 'Question',
})
prop_add(1, 0, {
    type: 'some_type_name',
    text: ' some virtual text',
    text_align: 'after',
})
```
---

[This one](https://github.com/vim/vim/issues/10786#issuecomment-1195339995) is not fixed yet. I guess we'll need to open a separate issue.

**Comment by brammool** [comment]
> Fixed in patch [9.0.0121](https://github.com/vim/vim/releases/tag/v9.0.0121):
> ```vim
> vim9script
> prop_type_add('some_type_name', {
>     highlight: 'Question',
> })
> prop_add(1, 0, {
>     type: 'some_type_name',
>     text: ' some virtual text',
>     text_align: 'after',
> })
> ```
> ---
> 
> [This one](https://github.com/vim/vim/issues/10786#issuecomment-1195339995) is not fixed yet. I guess we'll need to open a separate issue.

Does patch 9.0.0125 fix this?

-- 
hundred-and-one symptoms of being an internet addict:
216. Your pet rock leaves home.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by bfrg** [comment]
> Does patch 9.0.0125 fix this?

Unfortunately, no. The example above still doesn't work correctly.
